### ISSUE TITLE
Pooler error: login rejected after RELOAD

### ISSUE THREAD

**Issue by mrgtt** [pgbouncer/pgbouncer#297]
Using PgBouncer 1.8.1, after altering the configuration to require TLS everything went well but I've noticed that after a RELOAD PgBouncer refuses any login and require a full restart to accept again any connection.

Relevant changes in configuration:

`client_tls_sslmode = require`
`client_tls_ca_file = /etc/pgbouncer/server.crt`
`client_tls_key_file = /etc/pgbouncer/server.key`
`server_tls_sslmode = allow`
`auth_type = hba`
`auth_file = /etc/pgbouncer/userlist.txt`
`auth_hba_file = /dbroot/data/pg_hba.conf`
`auth_query = SELECT uname, phash from pga_user_lookup($1)`

The pg_hba.conf file is shared between Postgresql and PgBouncer, all connections from the network have ssl required except for local connections that are trusted.

Nothing useful from the log file, except for the error:

` WARNING C-0x108cff0: user/postgres@unix(46112):5432 Pooler Error: login rejected`

Is this a bug or am I missing something? Did anyone experienced somthing similar?

**Comment by PJMODOS** [comment]
It looks to me like you are connecting via unix socket, you can't use TLS over unix socket.

**Comment by mrgtt** [comment]
Local connections are trusted and does not use TLS as per pg_hba.conf:

    local   all             all                                     trust
    host    all             all             127.0.0.1/32            trust
    hostssl all             all             10.10.0.0/16           trust

Anyway here's the issue (on a CentOS 6):

    [pgbouncer@itmil-lnx-gdpr ~]$ psql -Upostgres
    psql (9.3.22)
    Type "help" for help.
    
    postgres=# \q
    [pgbouncer@itmil-lnx-gdpr ~]$ psql pgbouncer
    psql (9.3.22, server 1.8.1/bouncer)
    Type "help" for help.
    
    pgbouncer=# RELOAD;
    RELOAD
    pgbouncer=# \q
    [pgbouncer@itmil-lnx-gdpr ~]$ psql -Upostgres
    psql: ERROR:  login rejected
    [pgbouncer@itmil-lnx-gdpr ~]$ tail /var/log/pgbouncer/pgbouncer.log
    2018-06-14 08:58:43.365 2976 LOG C-0x15d70d0: postgres/postgres@unix(6267):5432 closing because: client close request (age=3)
    2018-06-14 08:59:03.767 2976 LOG C-0x15d70d0: (nodb)/pgbouncer@unix(6297):5432 pgbouncer access from unix socket
    2018-06-14 08:59:03.767 2976 LOG C-0x15d70d0: pgbouncer/pgbouncer@unix(6297):5432 login attempt: db=pgbouncer user=pgbouncer tls=no
    2018-06-14 08:59:05.627 2976 LOG Stats: 0 xacts/s, 0 queries/s, in 0 B/s, out 3 B/s, xact 0 us, query 0 us wait time 272106687693 us
    2018-06-14 08:59:08.165 2976 LOG RELOAD command issued
    2018-06-14 08:59:09.349 2976 LOG C-0x15d70d0: pgbouncer/pgbouncer@unix(6297):5432 closing because: client close request (age=5)
    2018-06-14 08:59:15.246 2976 LOG C-0x15d70d0: postgres/postgres@unix(6375):5432 login attempt: db=postgres user=postgres tls=no
    2018-06-14 08:59:15.246 2976 LOG C-0x15d70d0: postgres/postgres@unix(6375):5432 closing because: login rejected (age=0)
    2018-06-14 08:59:15.246 2976 LOG S-0x15dc0a0: postgres/pgbouncer@127.0.0.1:6432 closing because: unclean server (age=35)
    2018-06-14 08:59:15.246 2976 WARNING C-0x15d70d0: postgres/postgres@unix(6375):5432 Pooler Error: login rejected
    [pgbouncer@itmil-lnx-gdpr ~]$

Any suggestion will be appreciated

**Comment by DylanYoung** [comment]
Did you change those `auth_` settings? They require a restart...

**Comment by mrgtt** [comment]
After changing the auth_ settings a restart is done. Everything goes well until a RELOAD is needed (after a minor change like adding a database) then every connection is rejected and only a restart clear the issue.
This is annoying in production systems where adding a database will require a restart kicking off all the users.

**Comment by djester** [comment]
So the problem is not solved. A restart is still needed to apply the auth_ settings

**Comment by mrgtt** [comment]
Yes. Even in the latest version, I thought the fix for issue #220 in 1.10 could solve the problem but it's still there.
Simply put, RELOAD is of no use, we need to do a restart every time.
